### PR TITLE
Export pdf

### PR DESCRIPTION
**Pull request by Axedez**
[Trello Task](https://trello.com/_____)

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [ ] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
- [x] screenshot of feature is added if task has UI
image

### CODE REVIEW

File: app/assets/stylesheets/receipt.scss [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      display: block;
+      text-align: center;
+  }
+
+  .invoice-box table tr.information table td {
+      width: 100%;
+      display: block;
+      text-align: center;
+  }
+}
```
Comment (sviatoslav-krupa) [inline app/assets/stylesheets/receipt.scss:80]: add new line

File: app/views/account/utility_providers/show.html.slim [hunk 2]
```diff
@@ -0,0 +1,3 @@
+h1 = @utility_provider.contract_num
+= link_to "Show PDF", [:account, :utility_provider, format: :pdf],
+  class: 'btn btn-info btn-custom'
```
Comment (sviatoslav-krupa) [inline app/views/account/utility_providers/show.html.slim:3]: add new line

File: app/views/shared/account/_header.html.slim [hunk 3]
```diff
@@ -45,5 +45,5 @@ nav.navbar.navbar-expand-lg.navbar-dark.navbar-header
       li
         = link_to  destroy_user_session_path, method: :delete
           span
-            button.btn.btn-transparent.auth-btn.ml-2 
+            button.btn.btn-transparent.auth-btn.ml-2
               = t('.btn_destoy_auth')
```
Comment (sviatoslav-krupa) [inline app/views/shared/account/_header.html.slim:49]: add new line

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -13,7 +14,10 @@ def default_url_options(_options = {})
     { locale: I18n.locale }
   end
 
-  protected
+  def set_locale
```
Comment (sviatoslav-krupa) [inline app/controllers/application_controller.rb:17]: це відноситься до твоєї таски?

File: config/initializers/wicked_pdf.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+#
+# Use this to set up shared configuration options for your entire application.
+# Any of the configuration options shown here can also be applied to single
+# models by passing arguments to the `render :pdf` call.
+#
+# To learn more, check out the README:
+#
+# https://github.com/mileszs/wicked_pdf/blob/master/README.md
+
+WickedPdf.config = {
```
Comment (sviatoslav-krupa) [inline config/initializers/wicked_pdf.rb:11]: видали цей файл, він пустий

File: spec/controllers/home_controller_spec.rb [hunk 6]
```diff
@@ -7,7 +7,7 @@
   end
 
   describe 'render view #about' do
-    before { get :about }
+    subject { get :about }
```
Comment (sviatoslav-krupa) [inline spec/controllers/home_controller_spec.rb:10]: чому ти це змінив?

**Review by sviatoslav-krupa — CHANGES_REQUESTED** [review]

File: app/controllers/application_controller.rb (hunk 4 above)
Comment (Axedez) [reply app/controllers/application_controller.rb:17]: Це попередня таска, яка ще не змерджена

File: config/initializers/wicked_pdf.rb (hunk 5 above)
Comment (Axedez) [reply config/initializers/wicked_pdf.rb:11]: А якщо в майбутньому треба буде додати щось в конфіг? ЧИ в такому випадку просто створиться новий файл?

File: spec/controllers/home_controller_spec.rb (hunk 6 above)
Comment (Axedez) [reply spec/controllers/home_controller_spec.rb:10]: Прочитав на стеку + мейт вище також написав сабджект

Comment (Axedez) [reply spec/controllers/home_controller_spec.rb:10]: Прочитав на стеку + мейт вище також написав сабджект

File: app/assets/stylesheets/receipt.scss [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  border-bottom: 1px solid #ddd;
+  font-weight: bold;
+}
+
+.invoice-box table tr.details td {
+  padding-bottom: 20px;
+}
+
+.invoice-box table tr.item td{
+  border-bottom: 1px solid #eee;
```
Comment (pereslop) [inline app/assets/stylesheets/receipt.scss:57]: Duplicated style

File: app/assets/stylesheets/receipt.scss [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+@media only screen and (max-width: 600px) {
+  .invoice-box table tr.top table td {
+      width: 100%;
+      display: block;
+      text-align: center;
+  }
+
+  .invoice-box table tr.information table td {
```
Comment (pereslop) [inline app/assets/stylesheets/receipt.scss:76]: Dulicated style, the same as above, dry it

File: app/controllers/account/payments_controller.rb [hunk 9]
```diff
@@ -0,0 +1,26 @@
+class Account::PaymentsController < Account::AccountController
+  before_action :set_utility_providers
```
Comment (pereslop) [inline app/controllers/account/payments_controller.rb:2]: If you have only one action, you dont need it

File: app/views/account/payments/show.html.slim [hunk 10]
```diff
@@ -0,0 +1,3 @@
+h1 = @utility_provider.contract_num
+= link_to "Show PDF", [:account, :utility_provider, :payment, format: :pdf],
```
Comment (pereslop) [inline app/views/account/payments/show.html.slim:2]: I18n

File: app/views/account/utility_providers/_utility_provider.html.slim [hunk 11]
```diff
@@ -8,6 +8,11 @@ tr
   td
     span.cusom-center-in-table-example
       =< t(status_contract(utility_provider.is_active))
+  td.eye-btn.text-center
+    = link_to account_utility_provider_path(id: utility_provider),
+      method: :get
```
Comment (pereslop) [inline app/views/account/utility_providers/_utility_provider.html.slim:13]: Do really need it?

File: config/routes.rb [hunk 12]
```diff
@@ -12,9 +12,10 @@
 
     namespace :account do
       resources :users, except: %i[index destroy]
-      resources :utility_providers, only: %i[index new update] do
+      resources :utility_providers, only: %i[index new show update] do
         get 'search', on: :collection
         put 'disassociate', on: :member
+        resources :payments, only: %i[show]
```
Comment (pereslop) [inline config/routes.rb:18]: If you have only one action, you can just specify that one, you dont need recourses

File: spec/controllers/account/payments_controller_spec.rb [hunk 13]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe Account::PaymentsController, type: :controller do
+  let!(:utility_provider) { create(:billing_contract) }
```
Comment (pereslop) [inline spec/controllers/account/payments_controller_spec.rb:4]: Do you need to create  provider, if you dont check anything related to it in specs?

**Review by matsevich — APPROVED** [review]

**Review by marikrebega — APPROVED** [review]

**Review by Ukhanskyi — APPROVED** [review]

File: app/assets/stylesheets/receipt.scss [hunk 14]
```diff
@@ -0,0 +1,82 @@
+$white-blur: #eee;
```
Comment (BogdanBusko) [inline app/assets/stylesheets/receipt.scss:1]: move to file colors

File: app/controllers/account/payments_controller.rb [hunk 15]
```diff
@@ -0,0 +1,27 @@
+class Account::PaymentsController < Account::AccountController
+  def index; end
+
+  def show
+    @utility_provider = current_user.billing_contracts.find_by id: params[:id]
```
Comment (BogdanBusko) [inline app/controllers/account/payments_controller.rb:5]: enter after

File: app/controllers/account/payments_controller.rb [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+    if @utility_provider.present?
+      respond_to do |format|
+        format.html
+        format.pdf { render_pdf }
+      end
+    else
+      flash.now[:alert] = 'Contract not found!'
```
Comment (BogdanBusko) [inline app/controllers/account/payments_controller.rb:12]: mb you should have some redirect here

File: app/views/layouts/receipt_pdf.html.slim [hunk 17]
```diff
@@ -0,0 +1,10 @@
+doctype html
+html
+  head
+    = display_meta_tags site: "Contract #{@utility_provider.contract_num}"
+    = csrf_meta_tags
+    = csp_meta_tag
+    = wicked_pdf_stylesheet_link_tag 'receipt', media: 'all'
+
+  body.d-flex.flex-column.layout-body
+      = yield
```
Comment (BogdanBusko) [inline app/views/layouts/receipt_pdf.html.slim:10]: remove 2 spaces before =

File: spec/controllers/account/payments_controller_spec.rb [hunk 18]
```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe Account::PaymentsController, type: :controller do
+  let!(:user) { create(:user) }
+  let!(:utility_provider) { create(:billing_contract, user: user) }
+  let!(:wrong_utility_provider) { create(:billing_contract) }
+
+  before { sign_in user }
```
Comment (BogdanBusko) [inline spec/controllers/account/payments_controller_spec.rb:8]: mb you have some helper for this code? controller_macros

**Review by chu4 — APPROVED** [review]

**Pull request merged**